frontend/tests/e2e/newsletter.spec.ts

```diff
@@ -0,0 +1,65 @@
+/* eslint-disable testing-library/prefer-screen-queries */
+import { test, expect } from "@playwright/test";
+
+test.beforeEach(async ({ page }) => {
+  await page.goto("/newsletter");
+});
+
+test.afterEach(async ({ context }) => {
+  await context.close();
+});
+
+test("has title", async ({ page }) => {
+  await expect(page).toHaveTitle(/Newsletter | Simpler.Grants.gov/);
+});
+
+test("client side errors", async ({ page }) => {
+  await page.getByRole("button", { name: /subscribe/i }).click();
+
+  // Verify client-side errors for required fields
+  await expect(page.getByTestId("errorMessage")).toHaveCount(2);
+  await expect(page.getByText("Enter your first name.")).toBeVisible();
+  await expect(page.getByText("Enter your email address.")).toBeVisible();
+});
+
+test("successful signup", async ({ page }) => {
+  // TODO: determine if it's worth hitting this endpoint vs. mocking a specific response
```

Comment:
I'm curious what reviewers think about testing out the subscription endpoint with a valid-enough email address and the clean up associated with that (e.g. unsubscribing)

Reply:
Think it would be a good test to have because we need to refactor the frontend portion of the Next.js API endpoint to the new app router style. 

I believe a test written now that works should still work after we do our refactor. So the test is agnostic of implementation.

Reply:
Great, I agree that it'd be better to test the newsletter endpoint directly. In order to do so we'll need to add the env variables that start with `SENDY` as defined in `.env.development` to this workflow.

I don't have permissions to pursue either option, but here are two choices:

1. Use secrets in Github Actions (https://docs.github.com/en/actions/security-guides/using-secrets-in-github-actions). There's one potential downside mentioned in the docs: "With the exception of GITHUB_TOKEN, secrets are not passed to the runner when a workflow is triggered from a forked repository." That presumably limits open source contributors' access to the secrets and in that case it'll probably make sense to determine if there's an appropriate if/else statement that can be run in the test file to mock the endpoint in the case where the `SENDY` variables are empty strings.
2. Use variables in Github Actions (https://docs.github.com/en/actions/learn-github-actions/variables). My guess is that this won't be sufficient from a security perspective, but it's worth adding as an option since it's easier to implement.